Documentation update

menkej opened the pull request:
fix: documentation issues, first adressed in #141

Too many changes since that pull request so I decided to clean up the current README instead. Also #141 included other unrelated stuff and was stale, so I suggest replacing it with this change.

menkej commented:
I'm also thinking about extracting the contributing chapter into an own CONTRIBUTING.md and add clarify some things like tests, CHANGELOG  and all the other stuff I usually forget to do. ;-) What do you think about that?

AndreyNikiforov commented on README.md:
````diff
@@ -187,23 +169,27 @@ If you want to send notification emails using your Gmail account, and you have e
 
-    $ icloud --username jappleseed@apple.com --delete-from-keyring
+``` sh
+cloud --username jappleseed@apple.com --delete-from-keyring
````
should be `icloud` (missing "i")

menkej replied:
ok. Fixed that.

AndreyNikiforov commented on README.md:
```diff
@@ -309,8 +296,8 @@ branch. PRs should be based on the `pyicloud-ipd` branch and submitted to
 
 ### Building the Docker image
```
This section is duplicate of what is available above. Probably worth aggregating.

menkej replied:
I thought the same but did not want to rearrange the thing without discussion. Maybe it makes sense to put this somewhere separate as well in the future. We also have some FAQs already, p.e. problems on first run, how to handle directory-structure, etc. I thought this could be worth a wiki-page where others also might contribute, what do you think? Could be linked just here and updated outside releases.

AndreyNikiforov commented:
> I'm also thinking about extracting the contributing chapter into an own CONTRIBUTING.md and add clarify some things like tests, CHANGELOG and all the other stuff I usually forget to do. ;-) What do you think about that?

Separating contribution guidelines and development details sounds good to me.

menkej commented:
I added a new file CONTRIBUTING.md, inspired from http://contributor-covenant.org/version/1/4/ and added/updated contribution info in there. I was a little unsure about the Enforcement section. What do you think @AndreyNikiforov @ndbroadbent ?

menkej commented:
Hi @ndbroadbent !
The current top of README.md ("Community Maintained"...) reads a little like "this project is stalled... This is is currently not the case I guess as we have currently 2-4 people putting some time into this. I suggest removing that section to give (new) users a little better "first impression" and maybe add a reference to the CONTRIBUTING.md with a remark that we're always looking for additional help. What do you think?

ndbroadbent commented:
Hey @menkej, yeah I totally agree! Things are definitely different now, so please do feel free to remove that notice and update the README / CONTRIBUTING files

AndreyNikiforov commented:
> The current top of README.md ("Community Maintained"...) reads a little like "this project is stalled... This is is currently not the case I guess as we have currently 2-4 people putting some time into this. I suggest removing that section to give (new) users a little better "first impression" and maybe add a reference to the CONTRIBUTING.md with a remark that we're always looking for additional help. What do you think?

Totally agree! We need to provide accurate information and set expectations. I feel the following may be valuable to be mentioned on the first page:
- supported by a group of ppl (== community)
- by volunteers dedicating their free time (==no paid work)
- our values -- that is deeper topic, but we can mention at least our focus on quality with 100% test coverage and review process for PRs (and we can polish values section over time)
- release schedule (#198)

Then `how to use` section of course.

CONTRIBUTING as a separate doc with details on developing is a good approach. I think the first page may benefit from having `how to help` section and a link to contributing doc is just one of the ways to help. The reason is that testing, submitting ideas, helping to resolve issues, translating, etc -- all that is help for the project that does not require coding and may be much more valuable than actual coding.

@menkej that for driving this doc update effort!

menkej commented:
Hi @AndreyNikiforov , I addressed your remarks in a first shot.

I also added a badge for our current quality checks to README. We should discuss about the other (build and coverage).

I'd like to restructure CONTRIBUTING a little more in the next days, maybe. I like the structure of https://github.com/nektos/act/blob/master/CONTRIBUTING.md for example.

What do you think, is the current state worth a "next step"/merge?

menkej commented:
Oh, I forgot: I started a wiki page as a FAQ. I've never worked with the wiki on GitHub, so I dunno of the capabilities in terms of reliability/permissions, etc. This is not referenced in the README yet. What do you guys think about it?

https://github.com/icloud-photos-downloader/icloud_photos_downloader/wiki

AndreyNikiforov commented on CONTRIBUTING.md:
```diff
@@ -24,6 +24,7 @@ There are some requirements for pull requests:
   please update the [Usage](README.md#usage) section in the README.md.
 - Make sure your change is documented in the
 [Unreleased](CHANGELOG.md#unreleased) section in the CHANGELOG.md.
+- We aim to push out a Release once a week (Fridays),  if there is at least one new change in CHANGELOG.
```
Update frequency is important for user too, probably even more than for contributors IMO.
